### PR TITLE
Allow TLS13_AD_MISSING_EXTENSION for older versions

### LINKED ISSUE #15375: OpenSSL closes the connection when client drops psk_key_exchange_modes ext in renegotiation
When running current master (d5e08231dbef8) and 1.1.1-stable (7e12c2b3d9ccf97186e4) openssl just closes the connection when it receives a renegotiation ClientHello that drops the psk_key_exchange_modes extension compared to the initial ClientHello.

This is a regression compared to openssl-3.0.0-alpha16 and 1.1.1k.

Reproducer:
```
openssl req -x509 -newkey rsa -keyout /tmp/localhost.key -out /tmp/localhost.crt -subj /CN=localhost -nodes -batch
openssl s_server -key /tmp/localhost.key -cert /tmp/localhost.crt -www -client-renegotiation
```
And in separate terminal:
```
git clone https://github.com/tomato42/tlsfuzzer
pushd tlsfuzzer
git clone https://github.com/tlsfuzzer/tlslite-ng .tlslite-ng
ln -s .tlslite-ng/tlslite tlslite
git clone https://github.com/tlsfuzzer/python-ecdsa .python-ecdsa
ln -s .python-ecdsa/src/ecdsa ecdsa
PYTHONPATH=. python3 scripts/test-renegotiation-changed-clienthello.py -d
```
Result:
```
drop psk_key_exchange_modes in renegotiation ...
Error encountered while processing node ExpectServerHello(description='second handshake') (child: <tlsfuzzer.expect.ExpectCertificate object at 0x7eff4774f7f0>) with last message being: None
Error while processing
Traceback (most recent call last):
  File "/home/hkario/dev/tlsfuzzer/tlsfuzzer/runner.py", line 196, in run
    header, parser = self.state.msg_sock.\
  File "/home/hkario/dev/tlsfuzzer/tlslite/messagesocket.py", line 102, in recvMessageBlocking
    for res in self.recvMessage():
  File "/home/hkario/dev/tlsfuzzer/tlslite/messagesocket.py", line 84, in recvMessage
    for ret in self.recvRecord():
  File "/home/hkario/dev/tlsfuzzer/tlslite/recordlayer.py", line 893, in recvRecord
    for result in self._recordSocket.recv():
  File "/home/hkario/dev/tlsfuzzer/tlslite/recordlayer.py", line 210, in recv
    for record in self._recvHeader():
  File "/home/hkario/dev/tlsfuzzer/tlslite/recordlayer.py", line 144, in _recvHeader
    for result in self._sockRecvAll(1):
  File "/home/hkario/dev/tlsfuzzer/tlslite/recordlayer.py", line 131, in _sockRecvAll
    raise TLSAbruptCloseError()
tlslite.errors.TLSAbruptCloseError: TLSAbruptCloseError()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "scripts/test-renegotiation-changed-clienthello.py", line 408, in main
    runner.run()
  File "/home/hkario/dev/tlsfuzzer/tlsfuzzer/runner.py", line 222, in run
    raise AssertionError(
AssertionError: Unexpected closure from peer
```

`ec_point_formats` also fails, but that's tracked in #14875

### PR DESCRIPTION
**Pull request by kaduk**
Add a pass-through switch case for TLS13_AD_MISSING_EXTENSION in
ssl3_alert_code() and tls1_alert_code(), so that the call to
SSLfatal() in final_psk() will always actually generate an alert,
even for non-TLS1.3 protocol versions.

Fixes #15375

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

File: ssl/s3_enc.c
Comment (mattcaswell) [inline ssl/s3_enc.c:593]: Wouldn't it be better to map this to an actual SSLv3 alert like `SSL3_AD_HANDSHAKE_FAILURE`?

File: ssl/t1_enc.c
Comment (mattcaswell) [inline ssl/t1_enc.c:852]: Similarly here?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: ssl/s3_enc.c
Comment (kaduk) [reply ssl/s3_enc.c:593]: At present, the only way to get this alert is if you sent an extension first defined for TLS 1.3 and only useful for TLS 1.3, so you could reasonably be assumed to know what to do with the TLS 1.3 alert ... but I also don't actually care about this scenario.

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by kaduk** [comment]
I will wait until 24 hours after Matt's approval

**Review by paulidale — APPROVED** [review]

**Comment by kaduk** [comment]
Pushed to master; closing.
Thanks for the review!

**Pull request closed without merging**
